### PR TITLE
TRUNK-4645 "Thread.run() and Runnable.run() should not be called directl...

### PR DESCRIPTION
**Pull request by tomaszmueller**
...y"

### CODE REVIEW

**Comment by tomaszmueller** [comment]
@rkorytkowski https://issues.openmrs.org/browse/TRUNK-4645

File: api/src/main/java/org/openmrs/api/context/Daemon.java [hunk 1]
```diff
@@ -178,7 +178,7 @@ public void run() {
 				isDaemonThread.set(true);
 				try {
 					Context.openSession();
-					runnable.run();
+                    new Thread(runnable).start();
```
Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/api/context/Daemon.java]: This is actually intended as you can see the start method is called below in line 189: thread.start(). We need to annotate the run method (line 177) with: 
{code}
@SuppressWarnings("squid:3AS1217") //We intentionally do not start a new thread yet, rather wrap the run call in a session.
{code}

File: api/src/main/java/org/openmrs/api/context/Daemon.java [hunk 2]
```diff
@@ -271,7 +271,7 @@ public void run() {
...
```
Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/api/context/Daemon.java]: This is actually intended as you can see the start method is called below. Please revert it. We need to annotate the run method (line 270) with:

@SuppressWarnings("squid:3AS1217") //We intentionally do not start a new thread yet, rather wrap the run call in a session.

**Comment by tomaszmueller** [comment]
@rkorytkowski done

**Pull request merged**